File: algorithms/LinkedList.go
package main

import "fmt"

type Node struct {
	Value any
	Next  *Node
}

var head *Node

func InsertLinkedList(value any) {

	node := new(Node)
	node.Value = value
	node.Next = nil

	if head == nil {
		head = node
		return
	}
	temp := head
	for temp.Next != nil {
		temp = temp.Next
	}
	temp.Next = node
}

func DisplayLinkedList() {
	node := head

	for node != nil {
		fmt.Println(node.Value)
		node = node.Next
	}
}

File: DiveIntoSystems/LinkedList.c
#include "main.h"
#include <stdio.h>
#include <stdlib.h>

struct Node {
	int data;
	struct Node *next;
};

struct Node *head = NULL;

void insert(int data) {

	struct Node *NewNode = malloc(sizeof(struct Node));

	if (NewNode == NULL) {
		printf("Error initializing Node in Memory\n");
		return;
	}

	NewNode->data = data;
	NewNode->next = NULL;

	if (head == NULL) {
		head = NewNode;
		return;
	}

	struct Node *temp = head;

	while (temp->next != NULL) {
		temp = temp->next;
	}

	temp->next = NewNode;
}

void displayList(struct Node *head) {
	if (head == NULL) {
		return;
	}
	struct Node *temp = head;

	while (temp != NULL) {
		printf("%d \n", temp->data);
		temp = temp->next;
	}
	printf("\n");
}
